tests/test_evaluator.py

```diff
@@ -1 +1,162 @@
+        task_config = evaluator._prepare_task_configs(["ner", "sent"])
+        assert task_config == [NER, SENT]
+        task_config = evaluator._prepare_task_configs("sent")
+        assert task_config == [SENT]
+
+    def test_evaluate_single_raise_exception_model_not_found(
+        self, evaluator, non_existing_model_id
+    ):
+        evaluator.evaluation_config.testing = True
+        with pytest.raises(ModelDoesNotExistOnHuggingFaceHub):
+            evaluator._evaluate_single(
+                task_config=NER, model_id=[non_existing_model_id]
+            )
+
+    def test_evaluate_single_raise_exception_invalid_task(
+        self, evaluator, existing_model_id
+    ):
+        evaluator.evaluation_config.testing = True
+        with pytest.raises(InvalidArchitectureForTask):
+            evaluator._evaluate_single(task_config=NER, model_id=existing_model_id)
+
+    @pytest.mark.parametrize(
+        argnames="model_id, task_config, expected_results",
+        argvalues=[
+            (
+                "pin/senda",
+                SENT,
+                {
+                    "raw": [
+                        {"macro_f1": 1.0, "mcc": 0.5},
+                        {"macro_f1": 1.0, "mcc": 0.5},
+                    ],
+                    "total": {
+                        "macro_f1": 1.0,
+                        "macro_f1_se": 0.0,
+                        "mcc": 0.5,
+                        "mcc_se": 0.0,
+                    },
+                },
+            ),
+            (
+                "DaNLP/da-bert-tone-sentiment-polarity",
+                SENT,
+                {
+                    "raw": [
+                        {"macro_f1": 0.16666666666666666, "mcc": 0.25},
+                        {"macro_f1": 0.6666666666666666, "mcc": 0.25},
+                    ],
+                    "total": {
+                        "macro_f1": 0.41666666666666663,
+                        "macro_f1_se": 0.48999999999999994,
+                        "mcc": 0.25,
+                        "mcc_se": 0.0,
+                    },
+                },
+            ),
+        ],
+        ids=["sent_pin-senda", "sent_DaNLP-da-bert-tone-sentiment-polarity"],
+    )
+    def test_evaluate_single(self, evaluator, model_id, task_config, expected_results):
+        evaluator.evaluation_config.testing = True
+        evaluator._evaluate_single(task_config=task_config, model_id=model_id)
+        results = evaluator.evaluation_results[task_config.name][model_id]
+        assert expected_results == results
+
+    # TODO: Once we have more than one type of task, this should test combinations of tasks.
+    @pytest.mark.parametrize(
+        argnames="model_ids, tasks",
+        argvalues=[
+            (
+                ["pin/senda", "DaNLP/da-bert-tone-sentiment-polarity"],
+                ["sent", "sent"],
+            ),
+        ],
+        ids=["sent_pin-senda_sent_DaNLP"],
+    )
+    def test_evaluate(self, evaluator, model_ids, tasks):
```

Comment:
I would hardcode the combinations instead of using parametrize, as the individual tasks will all be evaluated in the `evaluate_single` tests anyway, so this should just check if we get the same thing, no?